cmd/argoexec/commands/agent.go

```diff
@@ -48,5 +51,15 @@ func initAgentExecutor() *executor.AgentExecutor {
 	if !ok {
 		log.Fatalf("Unable to determine workflow name from environment variable %s", common.EnvVarWorkflowName)
 	}
-	return executor.NewAgentExecutor(clientSet, restClient, config, namespace, workflowName)
+
+	var addresses []string
+	if err := json.Unmarshal([]byte(os.Getenv(common.EnvVarPluginAddresses)), &addresses); err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Does this mean we expect all agents to have plugin addresses? Even ones only used say for HTTP templates?

Reply:
this is implementation detail, but I wanted to keep the door open to remote plugins

Reply:
yes, it might be empty